src/base/torrentfileguard.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2016  Eugene Shalygin <eugene.shalygin@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+/// Utility class to defer file deletion
+
+#include <QString>
+
+class FileGuard
+{
+public:
+    FileGuard(const QString &path = QString());
+    ~FileGuard();
+
+    /// Cancels or re-establishes deferred file deletion
+    void setAutoRemove(bool remove) noexcept;
+
+private:
+    QString m_path;
+    bool m_remove;
+};
+
+/// reads settings for .torrent files from preferences
+/// and sets up file guard accordantly
+class TorrentFileGuard: public FileGuard
+{
+public:
+    TorrentFileGuard(const QString &path = QString());
+    ~TorrentFileGuard();
+    /// marks the torrent file as imported (added)
+    void markAsImported();
```

Comment:
@LordNyriox, @glassez: thank you for your suggestions! And the winner is... No, I just give up and propose `markAsAddedToSession()`. What do you think?